x-pack/test/custom_branding/tests/settings.ts

```diff
@@ -0,0 +1,100 @@
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const kibanaServer = getService('kibanaServer');
+  const esArchiver = getService('esArchiver');
+  const testSubjects = getService('testSubjects');
+  const find = getService('find');
+  const log = getService('log');
+  const PageObjects = getPageObjects(['settings', 'common', 'dashboard', 'timePicker', 'header']);
+
+  describe('custom branding', function describeIndexTests() {
+    const resetSettings = async () => {
+      const advancedSetting = await PageObjects.settings.getAdvancedSettings(
+        'xpackCustomBranding:pageTitle'
+      );
+      if (advancedSetting !== '') {
+        await PageObjects.settings.clearAdvancedSettings('xpackCustomBranding:pageTitle');
+      }
+      try {
+        await find.byCssSelector('img[alt="xpackCustomBranding:logo"]');
+        await PageObjects.settings.clearAdvancedSettings('xpackCustomBranding:logo');
+      } catch (e) {
+        log.debug('It is expected not to find custom branding properties set');
+      }
+      try {
+        await find.byCssSelector('img[alt="xpackCustomBranding:customizedLogo"]');
+        await PageObjects.settings.clearAdvancedSettings('xpackCustomBranding:customizedLogo');
+      } catch (e) {
+        log.debug('It is expected not to find custom branding properties set');
+      }
+    };
+
+    const goToSettings = async () => {
+      await PageObjects.settings.navigateTo();
+      await PageObjects.settings.clickKibanaGlobalSettings();
+    };
+
+    before(async function () {
+      await esArchiver.loadIfNeeded('test/functional/fixtures/es_archiver/logstash_functional');
+      await kibanaServer.uiSettings.replace({});
+      await PageObjects.settings.navigateTo();
+      await PageObjects.settings.clickKibanaGlobalSettings();
+      // clear settings before tests start
+      await resetSettings();
+    });
+
+    after(async function () {
+      await esArchiver.unload('test/functional/fixtures/es_archiver/logstash_functional');
+      await PageObjects.settings.navigateTo();
+      await PageObjects.settings.clickKibanaGlobalSettings();
+      await resetSettings();
+    });
+
+    beforeEach(async function () {
+      await goToSettings();
+    });
+
+    it('should allow setting custom page title through advanced settings', async function () {
+      const pageTitle = 'Custom Page Title';
+      const settingName = 'xpackCustomBranding:pageTitle';
+      await PageObjects.settings.setAdvancedSettingsInput(settingName, pageTitle);
+
+      const advancedSetting = await PageObjects.settings.getAdvancedSettings(settingName);
+      expect(advancedSetting).to.be(pageTitle);
+    });
+
+    it('should allow setting custom logo through advanced settings', async function () {
+      const settingName = 'xpackCustomBranding:logo';
+      await PageObjects.settings.setAdvancedSettingsImage(
+        settingName,
+        require.resolve('./mcdonalds_logo.png')
```

Comment:
__question:__ I'm far from a legal expert and likely exaggerating, but just curious, is it fine (no DMCA Takedown risk and such) to store copyrighted logo of the 3rd-party in our repository? :slightly_smiling_face:

Reply:
We should probably come up with a mock logo/name to use in the repo. I see ACME used in mockups regularly, we can come up some creative commons image to use for a logo or something.

Reply:
++ I'd avoid using anything copyrighted here

Reply:
I've seen some UIs re: custom branding use this wording:

> By uploading a custom company logo, I confirm I have the rights to use this image.

Should we consider using copy like that here?

Reply:
Thanks, updated this with acme logo.